<launch>
  <arg name="mav_name" default="firefly" />
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nbvplanner)/worlds/wind_turbine_capped.world" />
    <!-- <arg name="debug" value="true"/> -->
    <arg name="paused" value="true"/>
    <arg name="gui" value="true"/>
  </include>
  
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 firefly/vi_sensor/ground_truth fcu 1" />
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0.015 0.055 0.0065 -1.57 0.1 -1.57 fcu firefly/vi_sensor/camera_depth_optical_center_link 1" />
  <node pkg="tf" type="static_transform_publisher" name="link3_broadcaster" args="0.015 0.055 0.0065 -1.57 0.1 -1.57 fcu firefly/vi_sensor/camera_left_link 1" />
  <node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="0.015 -0.055 0.0065 -1.57 0.1 -1.57 fcu firefly/vi_sensor/camera_right_link 1" />
  
  <node name="rviz" pkg="rviz" type="rviz" output="screen"/>

  <node name="nbvPlanner" pkg="nbvplanner" type="nbvPlanner" output="screen" launch-prefix="gdb -ex run --args">
    <param name="tf_frame" type="string" value="world" />
    <param name="resolution" type="double" value="2.0" />
    <param name="use_history" type="bool" value="true" />
    <param name="sensor_max_range" type="double" value="20.0" />
    <param name="stl_file_path" type="string" value="$(find nbvplanner)/mesh/WindTurbine_Split_2220.stl" />
    <remap from="cam0/camera_info" to="/firefly/vi_sensor/left/camera_info"/>
    <remap from="cam1/camera_info" to="/firefly/vi_sensor/right/camera_info"/>
    <remap from="pointcloud" to="/firefly/vi_sensor/camera_depth/depth/points"/>
    <remap from="pose" to="firefly/odometry_sensor1/pose"/>
    <remap from="nbvplanner" to="firefly/nbvplanner"/>
    <remap from="octomap_full" to="/octomap_manager/get_map"/>
    <rosparam command="load" file="$(find nbvplanner)/resource/history.yaml" />
  </node>

  <group ns="$(arg mav_name)">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <arg name="model" value="$(find rotors_description)/urdf/$(arg mav_name)_vi_sensor.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="$(arg log_file)"/>
    </include>
    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_$(arg mav_name).yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/$(arg mav_name).yaml" />
      <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>
    <node name="exploration" pkg="nbvplanner" type="exploration" output="screen" >
      <rosparam command="load" file="$(find nbvplanner)/resource/history.yaml" />
    </node>
    
  </group>
  
</launch>
